### DISCUSSION TITLE
How to deploy based on roles in docker

### DISCUSSION THREAD

**Discussion by TaskDead** [apache/skywalking#11850] [Q&A] [answered]
I am trying to understand or find out if oap server can be deployed with roles  like receiver and aggregator in docker itself

**Comment by wu-sheng** [comment]
Why would you use cluster coordinator? Pure docker is not very suitable for product env.

**Reply by TaskDead** [reply]
I am not looking for a cluster , the network is such that I need to place the receiver near to the the agents and exporters, receiver can only connect with aggregator . I don't have direct connectivity where aggregator can be placed

**Reply by wu-sheng** [reply]
I don't think this is an official way. Receiver and aggregator are working as a cluster together remotely from the agent side, rather than edge server + backend server.

I think you misunderstood these concepts. 

But why do you deploy things like this? I am not clear about your intention.

**Reply by TaskDead** [reply]
There are three networks involved
1. Network A - Agent can be installed
2. Network B - Exporters can be installed
3. Network C- OAP setup

Network B can connect to A and C 
but 
Network C cannot connect to A and vice versa
So I was thinking having exporters and receiver in B
and aggregator in Network C

**Reply by wu-sheng** [reply, marked as answer]
If this is about network connectivity such as your company's security policy, you just need a kind of proxy, such as Nginx and Envoy to forward it.

**Reply by TaskDead** [reply]
Thank you
